### PR TITLE
Fix 'caml_unregister_frametable(table)' when 'table' is the first element

### LINKED ISSUE #11633: Segfault or infinite loop in 'caml_unregister_frametable'
Dear developers,

I am playing with JIT, compiling small codes, registering and unregistering frame tables.
Usually, when something goes wrong, it is because I messed with the runtime, but this time, it seems there is a possible bug in the function `caml_unregister_frametable`:
https://github.com/ocaml/ocaml/blob/679b5001e3f57560e8c6499c5ed7c54a1a63c422/runtime/roots_nat.c#L208-L229

For me, it does not take into account that the root of the list (`frametables`) can be the one we "unregister".
Thus, I would make the following changes at lines 221-228:
```c
if (frametables->data == table) {
  lnk = frametables;
  frametables = frametables->next;
  caml_stat_free(lnk); 
} else {
  iter_list(frametables->next,lnk) { 
     if(lnk->data == table) { 
       previous->next = lnk->next; 
       caml_stat_free(lnk); 
       break; 
     } 
     previous = lnk; 
   }
}
```
Regards,

### PR DESCRIPTION
**Pull request by recoules**
Closes #11633.

### CODE REVIEW

File: runtime/roots_nat.c [hunk 1]
```diff
@@ -218,13 +218,19 @@ void caml_unregister_frametable(intnat *table) {
...
-      caml_stat_free(lnk);
-      break;
+  if (frametables->data == table) {
+    lnk = frametables;
+    frametables = frametables->next;
+    caml_stat_free(lnk);
+  } else {
+    iter_list(frametables->next,lnk) {
+      if(lnk->data == table) {
+	previous->next = lnk->next;
```
Comment (smuenzel) [inline runtime/roots_nat.c]: I think this is mixing tabs and spaces for indentation, or at least it's rendering weird when I view it.

Comment (recoules) [reply runtime/roots_nat.c:228]: Should it be all spaces?

**Review by Ekdohibs — COMMENTED** [review]
This looks correct to me!

**Comment by gasche** [comment]
@Ekdohibs thanks! Do you know if this bug also requires a fix in 5.0?

(cc @Octachron: I'm happy to approve on @Ekdohibs' behalf, but can we include new fixes in 4.14 now that a RC is out?)

**Review by gasche — APPROVED** [review]

**Comment by Octachron** [comment]
We are indeed lacking a development branch for 4.14 right now. It would be better to merge the PR after the 4.14.1 release in less than two weeks. Then the fix will be part of the 4.14.2 release.

**Comment by gasche** [comment]
(If there is something to fix in trunk, possibly by just rebasing the proposed patch, we can merge it right now; ideally someone would do a separate PR for this.)

**Comment by Ekdohibs** [comment]
I'm not sure if there are any changes to do in trunk, since the logic for freeing frame descriptors changed quite a lot: https://github.com/ocaml/ocaml/blob/b06f2d1ac516ecab7b7ca5d7237434e8ee22aea2/runtime/frame_descriptors.c#L195-L208

**Review by nojb — COMMENTED** [review]
This needs a `Changes` entry I think.

**Pull request merged**